usr/src/cmd/bhyve/pci_virtio_block.c

```diff
@@ -80,10 +80,20 @@ _Static_assert(VTBLK_RINGSZ <= BLOCKIF_RING_MAX, "Each ring entry must be able t
 #define	VTBLK_BLK_ID_BYTES	20 + 1
 
 /* Capability bits */
-#define	VTBLK_F_SEG_MAX		(1 << 2)	/* Maximum request segments */
-#define	VTBLK_F_BLK_SIZE	(1 << 6)	/* cfg block size valid */
-#define	VTBLK_F_FLUSH		(1 << 9)	/* Cache flush support */
-#define	VTBLK_F_TOPOLOGY	(1 << 10)	/* Optimal I/O alignment */
+#define	VTBLK_F_BARRIER		(1 << 0)	/* Does host support barriers? */
+#define	VTBLK_F_SIZE_MAX	(1 << 1)	/* Indicates maximum segment size */
+#define	VTBLK_F_SEG_MAX		(1 << 2)	/* Indicates maximum # of segments */
+#define	VTBLK_F_GEOMETRY	(1 << 4)	/* Legacy geometry available  */
+#define	VTBLK_F_RO		(1 << 5)	/* Disk is read-only */
+#define	VTBLK_F_BLK_SIZE	(1 << 6)	/* Block size of disk is available*/
+#define	VTBLK_F_SCSI		(1 << 7)	/* Supports scsi command passthru */
+#define	VTBLK_F_FLUSH		(1 << 9)	/* Writeback mode enabled after reset */
+#define	VTBLK_F_WCE		(1 << 9)	/* Legacy alias for FLUSH */
+#define	VTBLK_F_TOPOLOGY	(1 << 10)	/* Topology information is available */
+#define	VTBLK_F_CONFIG_WCE	(1 << 11)	/* Writeback mode available in config */
+#define	VTBLK_F_MQ		(1 << 12)	/* Multi-Queue */
```

Comment:
Not sure -- it's that way in the original change (maybe a 1.2 draft??).  But since it's not being used anywhere, shouldn't hurt to pull it out.

Reply:
Hmm, I'm not sure what the best plan is.  Without good reason, I'd probably not bake in features that are only in a draft spec.  We can consider `#ifdef`ing it out rather than diverging.  Maybe this would be a good thing to bring up in the upstream review?

Reply:
I don't see it in the spec either but I find references to it via google and it's in the linux src:
https://github.com/torvalds/linux/blob/ac309e7744bee222df6de0122facaf2d9706fa70/include/uapi/linux/virtio_blk.h#L40

Reply:
I don't think there'd be any harm in leaving it out until it's codified somewhere. We can check with Allan, but I suspect there probably aren't any immediate plans to implement virtio-blk MQ support.. assuming it stays in it's current form, it can always be re-added w/o any problems.

Reply:
Found a rather short talk on it from 2014 https://www.linux-kvm.org/images/6/63/02x06a-VirtioBlk.pdf

Reply:
I looks like it'll make it's way into v1.2 of the spec: https://www.oasis-open.org/committees/ballot.php?id=3435 and was just missed..

Reply:
I'm fine with leaving it in.